chia/full_node/coin_store.py

```diff
@@ -449,10 +450,10 @@ async def get_coin_states_by_ids(
                 async with conn.execute(
                     f"SELECT confirmed_index, spent_index, coinbase, puzzle_hash, coin_parent, amount, timestamp "
                     f'FROM coin_record WHERE coin_name in ({"?," * (len(batch.entries) - 1)}?) '
-                    f"AND (confirmed_index>=? OR spent_index>=?)"
+                    f"AND (confirmed_index>=? OR spent_index>=?) AND (confirmed_index<=? AND spent_index<=?)"
```

Comment:
it's risky to make changes like these. the query planner might pick the wrong index for instance.
I would feel a bit better about it if this additional constraint would go away entirely when not used. Similar to `AND spent_index=0` below.
I suppose the parameters becomes a bit more complicated that way though.
maybe you can run the benchmark before and after this change to ensure it still performs well

Reply:
> I suppose the parameters becomes a bit more complicated that way though.

Changed but not sure what you mean with ^ ? What makes it more complicated? And which benchmark do you mean?

Reply:
I mean that, if you make `AND (confirmed_index<=? AND spent_index<=?)` conditional, you would also have to make it conditional whether to pass in the two arguments for the `?`.